engine/generic_list.hpp

```diff
@@ -182,7 +182,8 @@ class GenericList final : public Collection {
       }
       kvdk_assert(Collection::ExtractID(curr->Key()) == owner->ID(), "");
       kvdk_assert(
-          (curr->entry.meta.type == DataType && curr->Validate()) || dirty(),
+          (curr->entry.meta.mark.record_type == ElemType && curr->Validate()) ||
```

Comment:
Too many layers.
`record.type` is much simpler than `record.entry.meta.mark.record_type`

Reply:
This should be record.GetMark().type. We need to store record type in DataEntry. 
Maybe we don't need RecordMark, just two fields RecordType and RecordStatus

Reply:
Support to use RecordType and RecordStatus instead of RecordMark as its meaning is not much obvious.